packages/calypso-e2e/src/email-client.ts

```diff
@@ -124,6 +125,33 @@ export class EmailClient {
 		return null;
 	}
 
+	/**
+	 * Specialized method to return human-friendly magic login link.
+	 *
+	 * Also performs normalization of the link.
+	 * (eg. target calypso.live if run from that environment.)
+	 *
+	 * @param {Message} message Representing the message.
+	 * @returns {URL} URL object for the magic link.
+	 * @throws {Error} IF the message did not have any links.
+	 */
+	getMagicLink( message: Message ): URL {
+		const link = message.text?.links?.pop();
+
+		if ( ! link ) {
+			throw new Error( 'Message did not contain text links. ' );
+		}
+
+		const magicLinkURL = new URL( link?.href as string );
+		const baseURL = new URL( envVariables.CALYPSO_BASE_URL );
+
+		// Returns a new URL object with normalized magic link.
+		// Useful when running tests against environments other than the default
+		// CALYPSO_BASE_URL.
+		// Example: https://wordpress.com -> https://container-something.calypso.live.
+		return new URL( magicLinkURL.pathname + magicLinkURL.search, baseURL.origin );
```

Comment:
I'm doing something a bit odd here, and there must be an easier way to go about.

The idea here is to swap out the default `https://wordpress.com` origin with the one appropriate for the testing environment. For example, if we're testing on `http://calypso.localhost:3000`, when a test tries to open a link in the email **without** the normalization, it will load production.

The stumbling block for me has been that `URL` objects seem to be immutable, so once I create the initial `magicLinkURL` object, it isn't possible to modify its attributes.

```
> URL {
  origin: 'https://public-api.wordpress.com',

> a.origin = 'https://wercom'
'https://wercom'

> URL {
  origin: 'https://public-api.wordpress.com',
```

So, I create the two URL objects, then build a third by selecting the attributes I want from both.